runtime/lua/vim/shared.lua

```diff
@@ -450,6 +451,24 @@ function vim.tbl_add_reverse_lookup(o)
   return o
 end
 
+--- Return the inverse table with the keys and values swapped
+--- Examples:
+--- <pre>lua
+---   vim.tbl_inv({ a = 1, b = 2, c = 3 })
+---   -- == { [1] = 'a', [2] = 'b', [3] = 'c'}
+--- </pre>
+--- @param t table Table to invert
+--- @return table # Inverted table
+function vim.tbl_inv(t)
+  local r = {}
+  --- @diagnostic disable-next-line:no-unknown
+  for k, v in pairs(t) do
+    --- @diagnostic disable-next-line:no-unknown
+    r[v] = k
+  end
```

Comment:
What is table-specific about this? It could be streamed, so makes sense on `vim.iter`.

Reply:
What do you mean by table specific? This function will work with any table. I'm not sure how this would fit into `vim.iter`.

Reply:
If you prefer we can make this function private. It's just used in a few places in our codebase.